internal/pkg/app/routingTableManager.go

```diff
@@ -40,6 +40,31 @@ type LiveRouteWrapper struct {
 	FilterChain *filter.Chain
 }
 
+func (lrw *LiveRouteWrapper) Unregister(ctx context.Context, r *DefaultRoutingTableManager) error {
+	var e, err error
+	if lrw.Sender != nil {
+		err = r.pluginMgr.UnregisterSender(ctx, lrw.Sender)
+		if err != nil {
+			e = err
+		}
+	}
+	if lrw.Receiver != nil {
```

Comment:
Also, I think we should try to unregister receiver first because it gives the most penalty if left running?

Reply:
agreed, will do